components/tidb_query_expr/src/impl_cast.rs

```diff
@@ -604,6 +606,25 @@ fn cast_any_as_string<T: ConvertTo<Bytes> + Evaluable + EvaluableRet>(
     }
 }
 
+#[rpn_fn(nullable, capture = [ctx, extra])]
+#[inline]
+fn cast_year_as_string(
+    ctx: &mut EvalContext,
+    extra: &RpnFnCallExtra,
+    val: Option<&Int>,
+) -> Result<Option<Bytes>> {
+    match val {
+        None => Ok(None),
+        Some(val) => {
+            let val = match *val as i64 {
+                0 => "0000".as_bytes().to_vec(),
```

Comment:
you may use `b"0000".to_vec()` here.

Reply:
Nope, I figured out that I don't need to update tipb, we have FieldType in `get_cast_fn_rpn_meta` and it can be used to distinguish cast_year_to_int.